mongo/integration/unified/unified_spec_runner.go

```diff
@@ -282,9 +282,14 @@ func (tc *TestCase) Run(ls LoggerSkipper) error {
 		client.stopListeningForEvents()
 	}
 
-	for idx, expectedEvents := range tc.ExpectedEvents {
-		if err := verifyEvents(testCtx, expectedEvents); err != nil {
-			return fmt.Errorf("events verification failed at index %d: %v", idx, err)
+	// One of the bulkWrite spec tests expects update and updateMany to be grouped together into a single batch,
+	// but this isn't the case because of GODRIVER-1157. To work around this, we skip event verification for this test.
+	// This guard should be removed when GODRIVER-1157 is done.
+	if tc.Description != "BulkWrite on server that doesn't support arrayFilters with arrayFilters on second op" {
+		for idx, expectedEvents := range tc.ExpectedEvents {
+			if err := verifyEvents(testCtx, expectedEvents); err != nil {
+				return fmt.Errorf("events verification failed at index %d: %v", idx, err)
+			}
```

Comment:
This is an existing workaround from [`cmd_monitoring_helpers_test`](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/integration/cmd_monitoring_helpers_test.go#L223-L230) that is now being run in the unified test runner, so I had to port it over since GODRIVER-1157 is still unresolved.

I could have skipped the test (added to `skippedTestDescriptions`) or hardcoded in the correct, single expected event. Let me know if you think either of those would be better.

Reply:
Nice investigative work!

Aside: IIUC, GODRIVER-1157 is about merging updateOne / updateMany (and presumably deleteOne / deleteMany) into a single batch instead of separate batches. That may be worth clarifying in the description of GODRIVER-1157 when we do backlog grooming.